File: kernel/src/main/java/com/sos/scheduler/engine/kernel/cppproxy/DatabaseC.java
package com.sos.scheduler.engine.kernel.cppproxy;

import com.sos.scheduler.engine.cplusplus.runtime.CppProxy;
import com.sos.scheduler.engine.cplusplus.runtime.annotation.CppClass;


@CppClass(clas="sos::scheduler::Database", directory="scheduler", include="spooler.h")
public interface DatabaseC extends CppProxy {
    //Database.Type sisterType = new Database.Type();

    String db_name();       // Mit Kennwort
    String job_history_tablename();
}

File: engine-kernel/src/main/scala/com/sos/scheduler/engine/kernel/cppproxy/DatabaseC.scala
package com.sos.scheduler.engine.kernel.cppproxy

import com.sos.scheduler.engine.cplusplus.runtime.CppProxy
import com.sos.scheduler.engine.cplusplus.runtime.annotation.CppClass

@CppClass(clas = "sos::scheduler::Database", directory = "scheduler", include = "spooler.h")
trait DatabaseC extends CppProxy {

  def properties: Variable_setC // Mit "password"

  def jdbc_connection: AnyRef

  def transform_sql(statement: String): String
}
